PhotoStreamer/app/src/main/java/com/mtruehle/photostreamer/SearchFragment.java

```diff
@@ -0,0 +1,159 @@
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        RelativeLayout searchLayout = (RelativeLayout) inflater.inflate(R.layout.search_fragment,
+                container, false); // gets the layout identified by R.layout.search_fragment, sets that as the view for this. A way to "summon" the xml file layout thing, I believe.
+        Button searchButton = (Button) searchLayout.findViewById(R.id.search_button);
+        Button backButton = (Button) searchLayout.findViewById(R.id.back_button);
+        Button saveButton = (Button) searchLayout.findViewById(R.id.save_button);
+        Button nextButton = (Button) searchLayout.findViewById(R.id.next_button);
+        webView = (WebView) searchLayout.findViewById(R.id.results_webView);
+        helper = new FeedDbHelper(getContext());
+        currentlyDisplayedImage = -1;
+        webView.loadUrl("http://wisdomchasers.net/wp-content/uploads/2014/11/helloworld.gif");
+
+        searchButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                EditText searchField = (EditText) getActivity().findViewById(R.id.search_box);
+                String thingToSearch = searchField.getText().toString();
+                imageSearch(thingToSearch, false);
+                mostRecentQuery = thingToSearch;
+                Log.i("PRINTER", "testing");
+            }
+        });
+
+        backButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                goToPrevious();
+            }
+        });
+
+        saveButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                saveCurrent();
+            }
+        });
+
+        nextButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                goToNext();
+            }
+        });
+
+        handler = new HttpHandler(getActivity());
+        return searchLayout;
+    }
+
+    public void goToPrevious() {
+        if (currentlyDisplayedImage != -1) {
+            currentlyDisplayedImage = currentlyDisplayedImage - 1;
+            if (currentlyDisplayedImage < 0) {
+                currentlyDisplayedImage = 0;
+                Toast.makeText(getActivity(), "Already at beginning.", Toast.LENGTH_SHORT).show();
+            }
+            updateWebView();
+        }
+    }
+
+    public void goToNext() {
+        if (currentlyDisplayedImage != -1) {
+            currentlyDisplayedImage = currentlyDisplayedImage + 1;
+            if (currentlyDisplayedImage >= searchResults.size()) {
+                imageSearch(mostRecentQuery, true);
+            } else {
+                updateWebView();
+            }
+        }
+    }
+
+    public void updateWebView() {
```

Comment:
Whenever you see yourself duplicating code across classes it may be a good time to investigate if there is anotehr way you can structure your code to avoid it.
Duplicated code is bad because:
1. You have to write more code,
2. If you discover a subtle error in your implementation you now need to change it in multiple places and may forget about one of the places.